android/app/src/main/java/com/example/viewport/ViewportViewModel.kt

```
@@ -0,0 +1,252 @@
+    val user = repository.me()
+    val projects = repository.projects().projects
+    _uiState.update { it.copy(user = user, projects = projects, screen = OwnerScreen.Projects) }
+  }
+
+  private fun runAction(success: String? = null, block: suspend () -> Unit) {
+    viewModelScope.launch {
+      _uiState.update { it.copy(isLoading = true, error = null, message = null) }
+      try {
+        block()
+        _uiState.update { it.copy(isLoading = false, message = success) }
+      } catch (throwable: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)